pkg/controller/encryption/ipsec/ipsec_controller.go

```diff
@@ -0,0 +1,588 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name: localNodeName,
+		},
+		Spec: v1alpha1.KmeshNodeInfoSpec{
+			Name:       localNodeName,
+			Spi:        c.ipsecHandler.Spi,
+			Address:    []string{},
+			BootID:     c.localNode.Status.NodeInfo.BootID,
+			PodCirds:   c.localNode.Spec.PodCIDRs,
+			CreateTime: c.ipsecHandler.CreateTime,
+		},
+	}
+
+	for _, addr := range c.localNode.Status.Addresses {
+		if strings.Compare(string(addr.Type), string(v1.NodeInternalIP)) == 0 {
+			c.kmeshNodeInfo.Spec.Address = append(c.kmeshNodeInfo.Spec.Address, addr.Address)
+			c.ipsecHandler.SetNodeInfo(addr.Address, c.kmeshNodeInfo.Spec.BootID, c.ipsecHandler.Spi, c.ipsecHandler.CreateTime)
+		}
+	}
+
+	if err = c.attachTCToInternalNIC(); err != nil {
+		log.Errorf("failed to attach tc program to internal nic: %v", err)
+		return
+	}
+
+	defer c.queue.ShutDown()
+	go c.informer.Run(stop)
+	if !cache.WaitForCacheSync(stop, c.informer.HasSynced) {
+		log.Error("timed out waiting for caches to sync")
+		return
+	}
+
+	// create xfrm in rule, current host not update my kmeshnodeinfo
+	// the peer end does not use the key of the current host to send encrypted data.
+	if err := c.syncAllNodeInfo(); err != nil {
+		log.Errorf("failed to sync all node info: %v", err)
+		return
+	}
+
+	// update my kmesh node info, notify other machines that the key can be updated.
+	if err := c.updateLocalKmeshNodeInfo(); err != nil {
+		log.Errorf("failed to update local node info: %v", err)
+		return
+	}
+
+	c.ipsecHandler.StartWatch(c.UpdateXfrm)
+
+	go wait.Until(func() {
+		for c.processNextItem() {
+		}
+	}, 0, stop)
+
+	<-stop
+}
+
+func (c *IpsecController) handleOtherNodeInfo(node *v1alpha1.KmeshNodeInfo) error {
+	c.ipsecHandler.mutex.Lock()
+	defer c.ipsecHandler.mutex.Unlock()
+	for {
+		if node.Spec.Spi == c.ipsecHandler.Spi && strings.Compare(node.Spec.CreateTime, c.ipsecHandler.CreateTime) == 0 {
+			break
+		}
+
+		localNodeCreateTime, err := time.Parse(TimeFormatString, c.ipsecHandler.CreateTime)
+		if err != nil {
+			return fmt.Errorf("failed to parser local node ipsec create time! %v", err)
+		}
+
+		nodeSpiCreateTime, err := time.Parse(TimeFormatString, node.Spec.CreateTime)
+		if err != nil {
+			return fmt.Errorf("failed to parser local node ipsec create time! %v", err)
+		}
+
+		if localNodeCreateTime.Before(nodeSpiCreateTime) &&
+			time.Now().After(nodeSpiCreateTime.Add(time.Minute*5)) {
+			// There is a problem with the synchronization secret and the message cannot be processed
+			return fmt.Errorf("failed to get new secret, new spi is %v, create time is %v, my spi is %v, create time is %v",
+				node.Spec.Spi, nodeSpiCreateTime, c.ipsecHandler.Spi, c.ipsecHandler.CreateTime)
+		}
+
+		if localNodeCreateTime.After(nodeSpiCreateTime) {
+			if _, ok := c.ipsecHandler.oldIpSecKey[node.Spec.CreateTime]; !ok {
+				log.Info("can not found node spec ipsec key info, may be node not update, skip")
+				return nil
+			} else {
+				break
+			}
```

Comment:
I think this is very likely if you make one node disconnected for 5m. Then when it recovers connection, it cannot sync spi info correctly